framework/test/unit/modules/interoperability/mainchain/method.spec.ts

```diff
@@ -12,6 +12,106 @@
+import { EventQueue, MethodContext } from '../../../../../src/state_machine';
+import {
+	ChainAccount,
+	ChainAccountStore,
+	ChainStatus,
+} from '../../../../../src/modules/interoperability/stores/chain_account';
+import { OwnChainAccountStore } from '../../../../../src/modules/interoperability/stores/own_chain_account';
+
 describe('Mainchain Method', () => {
-	it.todo('should be able to create a new instance of MainchainInteroperabilityMethod');
+	const interopMod = new MainchainInteroperabilityModule();
+	const chainID = utils.intToBuffer(1, 4);
+	const chainAccountStoreMock = {
+		get: jest.fn(),
+		has: jest.fn(),
+	};
+	const ownchainAccountStoreMock = {
+		get: jest.fn(),
+	};
+	const interoperableCCMethods = new Map();
+	const defaultEventQueue = new EventQueue(0, [], [utils.hash(utils.getRandomBytes(32))]);
+	const timestamp = Date.now();
+	let mainchainInteroperabilityMethod: MainchainInteroperabilityMethod;
+	let methodContext: MethodContext;
+	let chainAccount: ChainAccount;
+
+	beforeEach(() => {
+		methodContext = createTransientMethodContext({ eventQueue: defaultEventQueue });
+		interopMod.stores.register(ChainAccountStore, chainAccountStoreMock as never);
+		interopMod.stores.register(OwnChainAccountStore, ownchainAccountStoreMock as never);
+		mainchainInteroperabilityMethod = new MainchainInteroperabilityMethod(
+			interopMod.stores,
+			interopMod.events,
+			interoperableCCMethods,
+			interopMod['internalMethod'],
+		);
+		chainAccount = {
+			name: 'chain',
+			lastCertificate: {} as never,
+			status: ChainStatus.ACTIVE,
+		};
+		ownchainAccountStoreMock.get.mockResolvedValue({ chainID: utils.intToBuffer(5, 4) });
+		chainAccountStoreMock.get.mockResolvedValue(chainAccount);
+		jest.spyOn(interopMod['internalMethod'], 'isLive').mockResolvedValue(true);
+	});
+
+	describe('isChannelActive', () => {
+		it('should return false when own chain', async () => {
+			ownchainAccountStoreMock.get.mockResolvedValue({ chainID });
+			const result = await mainchainInteroperabilityMethod.isChannelActive(
+				methodContext,
+				chainID,
+				timestamp,
+			);
+
+			expect(result).toBeFalse();
+		});
+
+		it('should return false if chain is not live and chain status is active', async () => {
+			jest.spyOn(interopMod['internalMethod'], 'isLive').mockResolvedValue(false);
+			const result = await mainchainInteroperabilityMethod.isChannelActive(
+				methodContext,
+				chainID,
+				timestamp,
+			);
+
+			expect(result).toBeFalse();
+		});
+
+		it('should return false if chain is live and chain status is not active', async () => {
+			chainAccountStoreMock.get.mockResolvedValue({
+				...chainAccount,
+				status: ChainStatus.TERMINATED,
+			});
+			const result = await mainchainInteroperabilityMethod.isChannelActive(
+				methodContext,
+				chainID,
+				timestamp,
+			);
+
+			expect(result).toBeFalse();
+		});
+
+		it('should return true if chain is live and chain status is not active', async () => {
```

Comment:
I suppose this should be:
```suggestion
		it('should return false if chain is live and chain status is not active', async () => {
```